pkg/controller/importconfig/klusterletconfighandler.go

```diff
@@ -69,3 +71,81 @@ func IndexManagedClusterByKlusterletconfigAnnotation(obj interface{}) ([]string,
 	}
 	return klusterletconfigs, nil
 }
+
+const (
+	KlusterletConfigBootstrapKubeConfigSecretsIndexKey = "klusterletconfig-bootstrapkubeconfig-secrets"
+)
+
+var _ handler.EventHandler = &enqueueManagedClusterByBootstrapKubeConfigSecrets{}
+
+// enqueueManagedClusterByBootstrapKubeConfigSecrets first finds the klusterletconfigs that using the secret, then
+// finds the managedclusters that using the klusterletconfigs.
+type enqueueManagedClusterByBootstrapKubeConfigSecrets struct {
+	// index klusterletconfig by the spec bootstrap kubeconfig secrets
+	klusterletconfigIndexer cache.Indexer
+
+	// index managedcluster by the annotation
+	managedclusterIndexer cache.Indexer
+}
+
+func (e *enqueueManagedClusterByBootstrapKubeConfigSecrets) Create(ctx context.Context,
+	evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	e.enqueue(evt.Object.GetName(), q)
+}
+
+func (e *enqueueManagedClusterByBootstrapKubeConfigSecrets) Update(ctx context.Context,
+	evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	e.enqueue(evt.ObjectNew.GetName(), q)
+}
+
+func (e *enqueueManagedClusterByBootstrapKubeConfigSecrets) Delete(ctx context.Context,
+	evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	e.enqueue(evt.Object.GetName(), q)
+}
+
+func (e *enqueueManagedClusterByBootstrapKubeConfigSecrets) Generic(ctx context.Context,
+	evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+	e.enqueue(evt.Object.GetName(), q)
+}
+
+func (e *enqueueManagedClusterByBootstrapKubeConfigSecrets) enqueue(secretName string, q workqueue.RateLimitingInterface) {
```

Comment:
can we have some rules like the secret must be in the managed cluster namespace, so here we can enqueue the secret.namespace directly?

Reply:
Not quit get the idea of enqueue the secret.namespace, may need to discuss offline.